connector-sdk/document/src/main/java/io/camunda/operation/IntrinsicOperation.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.operation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a method as an intrinsic operation. Positional parameters from the operation payload are
+ * passed to the method as arguments. The parameters are resolved by position, so the order of
+ * parameters in the method signature must match the order of parameters in the operation payload.
+ *
+ * <p>By default, all arguments are required. If an argument is nullable, it must be annotated with
+ * {@link javax.annotation.Nullable}.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface IntrinsicOperation {
+
+  String name();
+}
```

Comment:
We might want to use the term [IntrinsicFunction](https://en.wikipedia.org/wiki/Intrinsic_function) as it seems to be the canonical name for this concept. WDYT?